packages/flutter_tools/lib/src/windows/visual_studio.dart

```diff
@@ -167,6 +171,29 @@ class VisualStudio {
     ]);
   }
 
+  String? getCmakeVersion() {
+    final String _cmakeDocPath = _fileSystem.path.joinAll(<String>[
+      _usableVisualStudioDetails[_installationPathKey] as String,
```

Comment:
`_usableVisualStudioDetails` is defined as the information about an installed version that is capable of doing a Flutter Windows build. If it's already been set by the time this check is made, then it's too late. Notably, it will [bypass the build-time check](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/windows/build_windows.dart#L65-L69).

Having the right version of CMake needs to be a condition of setting `_usableVisualStudioDetails`, so the check can't use it.